lib/utils/ember.js

```diff
@@ -336,12 +331,50 @@ function isInjectedServiceProp(node, importedEmberName, importedInjectName) {
   );
 }
 
-function isInjectedControllerProp(node) {
-  return isPropOfType(node, 'controller');
+/**
+ * Checks if a node is a controller injection. Looks for:
+ * * controller()
+ * * Ember.inject.controller()
+ * @param {node} node
+ * @param {string} importedEmberName name that `Ember` is imported under
+ * @returns
+ */
+function isInjectedControllerProp(node, importedEmberName) {
+  return (
+    isPropOfType(node, 'controller') ||
+    (types.isProperty(node) &&
+      types.isCallExpression(node.value) &&
+      types.isMemberExpression(node.value.callee) &&
+      types.isMemberExpression(node.value.callee.object) &&
+      types.isIdentifier(node.value.callee.object.object) &&
+      node.value.callee.object.object.name === importedEmberName &&
+      types.isIdentifier(node.value.callee.object.property) &&
+      node.value.callee.object.property.name === 'inject' &&
+      types.isIdentifier(node.value.callee.property) &&
+      node.value.callee.property.name === 'controller')
+  );
 }
 
-function isObserverProp(node) {
-  return isPropOfType(node, 'observer');
+/**
+ * Checks if a node is an observer prop. Looks for:
+ * * observer()
+ * * Ember.observer()
+ * @param {node} node
+ * @param {string} importedEmberName name that `Ember` is imported under
+ * @returns
+ */
+function isObserverProp(node, importedEmberName) {
+  return (
+    isPropOfType(node, 'observer') ||
```

Comment:
Looks like this is missing an import check of the imported `observer` name (we could address as a follow-up).